crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -274,6 +270,39 @@ fn type_check_index_expression(
     }
 }
 
+fn get_expr_id(interner: &NodeInterner, stmt: &StmtId) -> Option<ExprId> {
+    match interner.statement(stmt) {
+        HirStatement::Expression(expr) => Some(expr),
+        HirStatement::Semi(expr) => Some(expr),
+        _ => None,
+    }
+}
+
+fn expected_semicolon(
+    interner: &NodeInterner,
+    stmt: &StmtId,
+    expr_type: &Type,
+    span: Span,
+) -> TypeCheckError {
+    let err = TypeCheckError::TypeMismatch {
+        expected_typ: Type::Unit.to_string(),
+        expr_typ: expr_type.to_string(),
+        expr_span: span,
+    };
+    match interner.statement(stmt) {
+        HirStatement::Semi(_) => err,
+        HirStatement::Expression(id) => {
+            match interner.expression(&id) {
+                HirExpression::Call(_) | HirExpression::MethodCall(_) => {
+                    err.add_context("Function calls in noir have no side effects. Perhaps you meant to bind the result to a name?")
```

Comment:
Function calls in noir do have side effect, we should allow users to call a function.

Reply:
I was under the impression that this error only happens if the function a user is calling has a return type but they have not assigned it to a variable. For example, in the tests where functions return `()`, the functions are still able to be called (https://github.com/noir-lang/noir/blob/21414180bd2f5a2d5f63bc75ee61a8d5e1a13aa6/crates/nargo/tests/test_data/global_consts/src/main.nr#L41).

Reply:
Yes, this happens with functions having a return value. But it should not be an error.